use-cases/flight-crew-scheduling/src/main/java/org/acme/flighcrewscheduling/domain/FlightAssignment.java

```diff
@@ -0,0 +1,116 @@
+package org.acme.flighcrewscheduling.domain;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.lookup.PlanningId;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@PlanningEntity
+public class FlightAssignment implements Comparable<FlightAssignment> {
+
+    // Needs to be kept consistent with equals on account of Employee's flightAssignmentSet, which is a SortedSet.
+    private static final Comparator<FlightAssignment> COMPARATOR = Comparator.comparing(FlightAssignment::getFlight)
+            .thenComparing(FlightAssignment::getIndexInFlight);
```

Comment:
We still need to sort the flight assignments, even if we remove `SortedSet`. It can be moved to the ` Employee` class and used by `org.acme.flighcrewscheduling.domain.Employee#countInvalidConnections`.